controllers/provider-aws/pkg/controlplane/internal/helper_test.go

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"encoding/json"
+	"testing"
+
+	apisaws "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws"
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	mockruntime "github.com/gardener/gardener-extensions/pkg/mock/apimachinery/runtime"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	namespace = "test"
+)
+
+func TestController(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "AWS Controlplane Internal Suite")
+}
+
+var _ = Describe("Helper", func() {
+	var (
+		ctrl *gomock.Controller
+
+		cp = &extensionsv1alpha1.ControlPlane{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "control-plane",
+				Namespace: namespace,
+			},
+			Spec: extensionsv1alpha1.ControlPlaneSpec{
+				ProviderConfig: &runtime.RawExtension{
+					Raw: encode(&apisaws.ControlPlaneConfig{
```

Comment:
Do you really have to use the `Raw` values here? Can't those `ComputeValues` functions take the unmarshalled values and operate on these? This would also fit the SRP.

Reply:
The `Raw` values are what the code under test decodes. Sure I could put there whatever and setup the decoder mocks to return concrete values, but I find encoding and decoding from JSON slightly more elegant, since the decoder mock setup in this case is always the same and very straightforward:

`decoder.EXPECT().Decode(cp.Spec.ProviderConfig.Raw, nil, &apisaws.ControlPlaneConfig{}).DoAndReturn(decode)`

Not sure what `ComputeValues` functions you refer to. If you mean `getConfigChartValues`, etc., these functions receive `ControlPlane` and decode it themselves, since the decoding itself is provider specific. I don't believe SRP is violated anywhere.